compose/config/config.py

```diff
@@ -493,19 +490,13 @@ def process_config_file(config_file, environment, service_name=None):
             config_file.get_networks(),
             'network',
             environment)
-        if config_file.version in (V3_1,):
-            processed_config['secrets'] = interpolate_config_section(
-                config_file,
-                config_file.get_secrets(),
-                'secrets',
-                environment
-            )
-    elif config_file.version == V1:
-        processed_config = services
+        processed_config['secrets'] = interpolate_config_section(
+            config_file,
+            config_file.get_secrets(),
+            'secrets',
+            environment)
```

Comment:
secrets is already checked against the schema during validation and the config version in `Config`, so I think we can avoid checking it again here.